File: kotlin_projects/04-2-One2NineAppWithRoom_kt2/app/src/main/java/com/example/one2nineappwithroom_kt2/compose/ScoreViewModel.kt
package com.example.one2nineappwithroom_kt2.compose

import androidx.compose.runtime.mutableStateListOf
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.one2nineappwithroom_kt2.repository.Score
import com.example.one2nineappwithroom_kt2.repository.ScoreRepository
import kotlinx.coroutines.launch

class ScoreViewModel(private val scoreRepository: ScoreRepository) : ViewModel() {

    var scores = mutableStateListOf<Score>()

    init {
        viewModelScope.launch {
            scoreRepository.getAllScores().collect {
                scores.clear()
                it.forEach { s -> scores.add(s) }
            }
        }
    }

    fun clearAllRecords() {
        viewModelScope.launch {
            scoreRepository.deleteAllScores()
        }
    }
}

File: flutter_projects/04-1-one2nineapp_fl/lib/views/scoreViewModel.dart

import 'package:flutter/material.dart';
import 'package:one2nineapp_fl/persistence/score.dart';
import 'package:one2nineapp_fl/persistence/scoreRepository.dart';

class ScoreViewModel extends ChangeNotifier {

  final ScoreRepository repository;

  ScoreViewModel(this.repository);

  Stream<List<Score>> getScoresStream() => repository.getAllScores();

  removeAllRecords() => repository.deleteAllScores();
}